vendor/models/base.py

```diff
@@ -25,12 +25,11 @@ def product_meta_default():
     return {'msrp':{'default':DEFAULT_CURRENCY, DEFAULT_CURRENCY: 0.00}}
 
 def validate_msrp(value):
-    if value['msrp']['default'] not in AVAILABLE_CURRENCIES:
-        raise ValidationError(_(f'Currency not available {key}'))
+    if value['msrp']['default'] not in AVAILABLE_CURRENCIES.keys():
+        raise ValidationError(_(f'Currency not available'))
 
-    for key in [ key for key in value['msrp'].keys() if key != 'default']:
-        if key not in AVAILABLE_CURRENCIES:
-            raise ValidationError(_(f'Currency not available {key}'))
+    if not set(AVAILABLE_CURRENCIES.keys()).intersection(value['msrp'].keys()):
```

Comment:
does `value['msrp'].keys()` need to be type converted to a set() too for the intersect?

Reply:
Please write any tests necessary to confirm this works.  Moving it to a shared function would make testing easier.

Reply:
Added test for products, offers and validator to user is_curency_available in the model.utils file

Reply:
I was more asking because all the docs that use intersection actually pass in sets as arguments.  If lists and dict.keys() work, this is fine.